From 86e37c1c09c23924c4e055a3d4b8c79f19cd0599 Mon Sep 17 00:00:00 2001
From: Maximilian Bosch <maximilian@mbosch.me>
Date: Mon, 10 Aug 2020 21:33:39 +0200
Subject: [PATCH] Remove references to dropped `sys_nerr` & `sys_errlist` for
 `glibc-2.32` compat

According to the release-notes[1], `strerror(3)` should be used. This is
already the case, however the source tries to be backwards-compatible by
supporting `sys_nerr` & `sys_errlist` which breaks compilation
unfortunately.

Simply using `strerror` fixes the problems.

[1] https://sourceware.org/pipermail/libc-announce/2020/000029.html
---
 utils.c | 12 +-----------
 1 file changed, 1 insertion(+), 11 deletions(-)

diff --git a/utils.c b/utils.c
index 3ec70b6..430f027 100644
--- a/utils.c
+++ b/utils.c
@@ -2003,7 +2003,6 @@ int n;
 
 #ifdef HAVE_SYS_ERRLIST
     extern char *sys_errlist[];
-    extern int sys_nerr;
 #endif
 
 /*
@@ -2019,16 +2018,7 @@ int errnum;
     sprintf(buf, "Error %d", errnum);
     return buf;
 #else
-    if (errnum < 0 || errnum > sys_nerr) {
-	sprintf(buf, "Error %d (!)", errnum);
-	return buf;
-    } else {
-#ifdef HAVE_STRERROR
-	return strerror(errnum);
-#else
-	return sys_errlist[errnum];
-#endif
-    }
+    return strerror(errnum);
 #endif
 }
 
-- 
2.25.4

